src/datasets/utils/py_utils.py

```diff
@@ -657,7 +659,11 @@ def _save_code(pickler, obj):
     #
     # Only those two lines are different from the original implementation:
     co_filename = (
-        "" if obj.co_filename.startswith("<") or obj.co_name == "<lambda>" else os.path.basename(obj.co_filename)
+        ""
+        if obj.co_filename.startswith("<")
+        or obj.co_filename.split(os.path.sep)[-2].startswith("ipykernel_")
```

Comment:
In case the co_filename doesn't have a parent dir:
```suggestion
        or (len(obj.co_filename.split(os.path.sep)) > 1 and obj.co_filename.split(os.path.sep)[-2].startswith("ipykernel_"))
```